### ISSUE TITLE
`environment.yml` is OS-specific

### ISSUE THREAD

**Issue by astrojuanlu** [iaguerri/climate_change_cities#1]
I tried recreating the conda environment with micromamba on Linux, but unfortunately it's failing to locate some packages:

```
$ micromamba create -f environment.yml -n climate

                                           __
          __  ______ ___  ____ _____ ___  / /_  ____ _
         / / / / __ `__ \/ __ `/ __ `__ \/ __ \/ __ `/
        / /_/ / / / / / / /_/ / / / / / / /_/ / /_/ /
       / .___/_/ /_/ /_/\__,_/_/ /_/ /_/_.___/\__,_/
      /_/

Found conda-prefix at '/home/juanlu/.micromamba/envs/climate'. Overwrite?: [y/N] y
pyviz/linux-64                                       1.7kB @   2.5kB/s  0.7s
pyviz/noarch                                        25.1kB @  34.7kB/s  0.7s
anaconda/noarch                                    440.9kB @ 526.1kB/s  0.8s
anaconda/linux-64                                    2.3MB @   2.7MB/s  0.9s
pkgs/main/noarch                                   815.5kB @ 610.0kB/s  0.5s
pkgs/r/noarch                                        1.3MB @ 729.9kB/s  0.9s
pkgs/r/linux-64                                      1.4MB @ 790.1kB/s  1.1s
pkgs/main/linux-64                                   4.7MB @   1.6MB/s  1.6s
conda-forge/noarch                                   9.0MB @   2.1MB/s  3.8s
conda-forge/linux-64                                24.8MB @   3.2MB/s  7.8s
Encountered problems while solving:
  - nothing provides requested anyio 3.6.1 py310h5588dad_0
  - nothing provides requested argon2-cffi-bindings 21.2.0 py310he2412df_2
  - nothing provides requested blosc 1.21.1 h74325e0_3
  - nothing provides requested bokeh 2.4.3 py310h5588dad_0
  - nothing provides requested boost-cpp 1.74.0 h9f4b32c_8
  - nothing provides requested bottleneck 1.3.5 py310h2873277_0
  - nothing provides requested brotli 1.0.9 h8ffe710_7
  - nothing provides requested brotli-bin 1.0.9 h8ffe710_7
  - nothing provides requested brotlipy 0.7.0 py310he2412df_1004
  - nothing provides requested bzip2 1.0.8 h8ffe710_4
  - nothing provides requested ca-certificates 2022.6.15 h5b45459_0
  - nothing provides requested cairo 1.16.0 h0ac17fb_1011
  - nothing provides requested cartopy 0.20.3 py310hfbfc819_1
  - nothing provides requested certifi 2022.6.15 py310h5588dad_0
  - nothing provides requested cffi 1.15.1 py310hcbf9ad4_0
  - nothing provides requested cfitsio 4.1.0 h5a969a9_0
  - nothing provides requested cftime 1.6.1 py310h2873277_0
  - nothing provides requested click 8.1.3 py310h5588dad_0
  - nothing provides requested cryptography 37.0.1 py310h21b164f_0
  - nothing provides requested curl 7.83.1 h789b8ee_0
  - nothing provides requested cytoolz 0.12.0 py310he2412df_0
  - nothing provides requested debugpy 1.5.1 py310hd77b12b_0
  - nothing provides requested entrypoints 0.4 py310haa95532_0
  - nothing provides requested expat 2.4.8 h39d44d4_0
  - nothing provides requested fiona 1.8.21 py310hfed10a5_2
  - nothing provides requested fontconfig 2.14.0 hce3cb01_0
  - nothing provides requested fonttools 4.34.4 py310he2412df_0
  - nothing provides requested freetype 2.10.4 h546665d_1
  - nothing provides requested freexl 1.0.6 ha8e266a_0
  - nothing provides requested gdal 3.5.1 py310h2a306c7_1
  - nothing provides requested geos 3.11.0 h39d44d4_0
  - nothing provides requested geotiff 1.7.1 h714bc5f_3
  - nothing provides requested gettext 0.19.8.1 ha2e2712_1008
  - nothing provides requested hdf4 4.2.15 h0e5069d_3
  - nothing provides requested hdf5 1.12.1 nompi_h57737ce_104
  - nothing provides requested icu 70.1 h0e60522_0
  - nothing provides requested importlib-metadata 4.11.4 py310h5588dad_0
  - nothing provides requested intel-openmp 2022.1.0 h57928b3_3787
  - nothing provides __win needed by ipykernel-6.15.1-pyh025b116_0
  - nothing provides requested ipython 8.3.0 py310haa95532_0
  - nothing provides requested jedi 0.18.1 py310haa95532_1
  - nothing provides requested jpeg 9e h8ffe710_2
  - nothing provides requested jupyter_client 7.2.2 py310haa95532_0
  - nothing provides requested jupyter_core 4.10.0 py310haa95532_0
  - nothing provides requested kealib 1.4.15 h8995ca9_0
  - nothing provides requested kiwisolver 1.4.4 py310h476a331_0
  - nothing provides requested krb5 1.19.3 hc8ab02b_0
  - nothing provides requested lcms2 2.12 h2a16943_0
  - nothing provides requested lerc 3.0 h0e60522_0
  - nothing provides requested libblas 3.9.0 15_win64_mkl
  - nothing provides requested libbrotlicommon 1.0.9 h8ffe710_7
  - nothing provides requested libbrotlidec 1.0.9 h8ffe710_7
  - nothing provides requested libbrotlienc 1.0.9 h8ffe710_7
  - nothing provides requested libcblas 3.9.0 15_win64_mkl
  - nothing provides requested libcurl 7.83.1 h789b8ee_0
  - nothing provides requested libdeflate 1.12 h8ffe710_0
  - nothing provides requested libffi 3.4.2 h8ffe710_5
  - nothing provides requested libgdal 3.5.1 h9e25771_1
  - nothing provides requested libglib 2.72.1 h3be07f2_0
  - nothing provides requested libiconv 1.16 he774522_0
  - nothing provides requested libkml 1.3.0 h9859afa_1014
  - nothing provides requested liblapack 3.9.0 15_win64_mkl
  - nothing provides requested libnetcdf 4.8.1 nompi_h1cc8e9d_102
  - nothing provides requested libpng 1.6.37 h1d00b33_3
  - nothing provides requested libpq 14.4 h1ea2d34_0
  - nothing provides requested librttopo 1.1.0 h2842628_11
  - nothing provides requested libspatialindex 1.9.3 h39d44d4_4
  - nothing provides requested libspatialite 5.0.1 ha17912d_18
  - nothing provides requested libssh2 1.10.0 h9a1e1f7_2
  - nothing provides requested libtiff 4.4.0 h2ed3b44_1
  - nothing provides requested libwebp 1.2.3 h8ffe710_0
  - nothing provides requested libwebp-base 1.2.3 h8ffe710_0
  - nothing provides requested libxcb 1.13 hcd874cb_1004
  - nothing provides requested libxml2 2.9.14 hf5bbc77_3
  - nothing provides requested libzip 1.9.2 h519de47_0
  - nothing provides requested libzlib 1.2.12 h8ffe710_2
  - nothing provides requested llvmlite 0.38.1 py310h2c03ce5_0
  - nothing provides requested lz4 4.0.0 py310h7b86b54_2
  - nothing provides requested lz4-c 1.9.3 h8ffe710_1
  - nothing provides requested m2w64-gcc-libgfortran 5.3.0 6
  - nothing provides requested m2w64-gcc-libs 5.3.0 7
  - nothing provides requested m2w64-gcc-libs-core 5.3.0 7
  - nothing provides requested m2w64-gmp 6.1.0 2
  - nothing provides requested m2w64-libwinpthread-git 5.0.0.4634.697f757 2
  - nothing provides requested markupsafe 2.1.1 py310he2412df_1
  - nothing provides requested matplotlib-base 3.5.2 py310h79a7439_0
  - nothing provides requested mistune 0.8.4 py310he2412df_1005
  - nothing provides requested mkl 2022.1.0 h6a75c08_874
  - nothing provides requested msgpack-python 1.0.4 py310h476a331_0
  - nothing provides requested msys2-conda-epoch 20160418 1
  - nothing provides requested nb_conda_kernels 2.3.1 py310h5588dad_1
  - nothing provides requested nest-asyncio 1.5.5 py310haa95532_0
  - nothing provides requested netcdf4 1.6.0 nompi_py310h5489b47_100
  - nothing provides requested numba 0.55.2 py310h77579ad_0
  - nothing provides requested numpy 1.22.3 py310hed7ac4c_2
  - nothing provides requested openjpeg 2.4.0 hb211442_1
  - nothing provides requested openssl 3.0.5 h8ffe710_0
  - nothing provides requested pandas 1.4.3 py310hf5e1058_0
  - nothing provides requested pandoc 2.18 h57928b3_0
  - nothing provides requested pcre 8.45 h0e60522_0
  - nothing provides requested pillow 9.2.0 py310h767b3fd_0
  - nothing provides requested pixman 0.40.0 h8ffe710_0
  - nothing provides requested poppler 22.04.0 h24fffdf_1
  - nothing provides requested postgresql 14.4 he353ca9_0
  - nothing provides requested proj 9.0.1 h1cfcee9_1
  - nothing provides requested psutil 5.9.1 py310he2412df_0
  - nothing provides requested pthread-stubs 0.4 hcd874cb_1001
  - nothing provides requested pyproj 3.3.1 py310h843b76b_1
  - nothing provides requested pyrsistent 0.18.1 py310he2412df_1
  - nothing provides requested pysocks 1.7.1 py310h5588dad_5
  - nothing provides requested python 3.10.5 hcf16a7b_0_cpython
  - nothing provides requested pywin32 302 py310h2bbff1b_2
  - nothing provides requested pywinpty 2.0.6 py310h00ffb61_0
  - nothing provides requested pyyaml 6.0 py310he2412df_4
  - nothing provides requested pyzmq 22.3.0 py310hd77b12b_2
  - nothing provides requested rasterio 1.3.0 py310h8cfb04a_0
  - nothing provides requested rtree 1.0.0 py310h1cbd46b_1
  - nothing provides requested scikit-learn 1.1.1 py310h4dafddf_0
  - nothing provides requested scipy 1.8.1 py310h33db832_0
  - nothing provides requested setuptools 63.2.0 py310h5588dad_0
  - nothing provides requested shapely 1.8.2 py310h523c7d9_3
  - nothing provides requested snappy 1.1.9 h82413e6_1
  - nothing provides requested sniffio 1.2.0 py310h5588dad_3
  - nothing provides requested sqlite 3.39.1 h8ffe710_0
  - nothing provides requested statsmodels 0.13.2 py310h2873277_0
  - nothing provides requested tbb 2021.5.0 h2d74725_1
  - nothing provides requested terminado 0.15.0 py310h5588dad_0
  - nothing provides requested tiledb 2.9.5 h3132609_0
  - nothing provides requested tk 8.6.12 h8ffe710_0
  - nothing provides requested tornado 6.1 py310h2bbff1b_0
  - nothing provides requested ucrt 10.0.20348.0 h57928b3_0
  - nothing provides requested unicodedata2 14.0.0 py310he2412df_1
  - nothing provides requested vc 14.2 hb210afc_6
  - nothing provides requested vs2015_runtime 14.29.30037 h902a5da_6
  - nothing provides requested win_inet_pton 1.1.0 py310h5588dad_4
  - nothing provides requested winpty 0.4.3 4
  - nothing provides requested xerces-c 3.2.3 h0e60522_5
  - nothing provides requested xorg-libxau 1.0.9 hcd874cb_0
  - nothing provides requested xorg-libxdmcp 1.1.3 hcd874cb_0
  - nothing provides requested xz 5.2.5 h62dcd97_1
  - nothing provides requested yaml 0.2.5 h8ffe710_2
  - nothing provides requested zlib 1.2.12 h8ffe710_2
  - nothing provides requested zstd 1.5.2 h6255e5f_2

The environment can't be solved, aborting the operation
error    libmamba Could not solve for environment specs
critical libmamba UnsatisfiableError
```

I suspect the reason is that `environment.yml` was generated using `conda env export` from a Windows machine:

https://github.com/iaguerri/climate_change_cities/blob/4549dc117ebaa9b670ac4415aee853360623b0f8/environment.yml#L280

Some background: https://docs.readthedocs.io/en/stable/guides/conda.html#creating-the-environment-yml

> There are several ways of [exporting a conda environment](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#sharing-an-environment):
> 
> - `conda env export` will produce a complete list of all the packages installed in the environment with their exact versions. This is the best option to ensure reproducibility, but can create problems if done from a different operative system than the target machine, in our case Ubuntu Linux.
> - `conda env export --from-history` will only include packages that were explicitly requested in the environment, excluding the transitive dependencies. This is the best option to maximize cross-platform compatibility, however it may include packages that are not needed to build your docs.
> - And finally, you can also write it by hand. This allows you to pick exactly the packages needed to build your docs (which also results in faster builds) and overcomes some limitations in the conda exporting capabilities.

In summary, it would be nice to provide an `environment.yml` only with non-pinned, explicit dependencies, like

```
name: climate_change
channels:
  - conda-forge
dependencies:
  - python=3.10
  - geopandas
  - cartopy
  - xarray
  - rioxarray
  - shapely
```

**Comment by iaguerri** [comment]
New ```environment.yml``` uploaded

**Comment by astrojuanlu** [comment]
I see it's in https://github.com/iaguerri/climate_change_cities/tree/develop, thanks!
